packages/nextcloud/lib/src/webdav/webdav_client.dart

```diff
@@ -355,40 +390,33 @@ class WebDavClient {
   ///
   /// See:
   ///   * [get_Request] for the request sent by this method.
-  Stream<List<int>> getStream(
+  Future<Stream<List<int>>> getStream(
     PathUri path, {
     void Function(double progress)? onProgress,
-  }) {
+  }) async {
     final request = get_Request(path);
-    // ignore: discarded_futures
-    final response = csrfClient.send(request);
-    final controller = StreamController<List<int>>();
+    final streamedResponse = await csrfClient.send(request);
+    if (streamedResponse.statusCode != 200) {
+      final response = await http.Response.fromStream(streamedResponse);
+      throw DynamiteStatusCodeException(response);
+    }
 
-    unawaited(
-      response.then(
-        (response) {
-          final contentLength = response.contentLength;
-          var downloaded = 0;
-
-          response.stream.listen(
-            (chunk) async {
-              controller.add(chunk);
-              downloaded += chunk.length;
-              if (contentLength != null) {
-                onProgress?.call(downloaded / contentLength);
-              }
-            },
-            onDone: () {
-              onProgress?.call(1);
-              controller.close();
-            },
-          );
-        },
-        // ignore: avoid_types_on_closure_parameters
-        onError: (Object error) {
-          controller.addError(error);
-        },
```

Comment:
Why did you remove the on error callback?

Reply:
This was needed so the error would appear in the stream. The function is now async and the error will happen on the await and not in the stream. You can check the test for this, it still works.